eng/pipelines/templates/steps/build-test.yml

```diff
@@ -60,8 +60,9 @@ steps:
 
   - ${{ if eq(parameters.TestProxy, true) }}:
     - pwsh: |
-        # $(Build.SourcesDirectory)/test-proxy.log is the hardcoded output log location for the test-proxy-tool.yml
-        cat $(Build.SourcesDirectory)/test-proxy.log
+        # ambitious_azsdk_test_proxy is the hardcoded container name used
+        # by the test proxy startup script
+        docker logs ambitious_azsdk_test_proxy
```

Comment:
@seankane-msft https://github.com/Azure/azure-sdk-tools/issues/2539 is still open and I haven't had time to get to it. I disagree with re-enabling docker here, and you're seeing a bunch of builds fail already. 

You're already seeing a bunch of failures unfortunately.

Reply:
#16921 is reverting the change